core/controllers/improvements.py

```diff
@@ -27,6 +27,31 @@
 from core.domain import exp_fetchers
 from core.domain import improvements_domain
 from core.domain import improvements_services
+from core.domain import user_services
+
+
+def get_task_dict_with_username_and_profile_picture(task_entry):
+    """Returns a task entry dict with the username and profile picture
+    URL inserted.
+
+    Args:
+        task_entry: improvements_domain.TaskEntry. The TaskEntry domain object.
+
+    Returns:
+        TaskEntryDict. TaskEntry dict username and profile picture
+        URL inserted.
+    """
+
+    task_entry_dict = task_entry.to_dict()
+    if task_entry.resolver_id:
+        resolver_settings = user_services.get_user_settings(
+                task_entry.resolver_id, strict=True) # type: ignore[no-untyped-call]
+        if resolver_settings:
```

Comment:
Is this `if` needed? Because above you have `strict=True`

Reply:
Yes you are right, it's not needed.